students/luftsmeerflier/lesson04/mailroom_3.py

```diff
@@ -0,0 +1,164 @@
+#/usr/bin/env python3
+import sys
+
+donors = [
+	#The actual donations themselves are a list of objects: Donations.date and Donations.amount
+	{
+		"first_name": "John",
+		"last_name": "Quigley",
+		"address": "555 Camelot RD N London, Ontario",
+		"donations": 
+			[
+				("2019-02-04", 5000.00),
+				("2019-02-04", 4250.00),
+				("2019-02-04", 3000.00)
+		
+			]
+	},
+	{
+		"first_name": "Sara",
+		"last_name": "Smith",
+		"address": "256 Camden RD N London, UK",
+		"donations": [
+			("2020-04-02", 2000.00),
+			("2019-02-04", 42000.00),
+			("2019-02-04", 32000.00)
+		]
+	},
+	{
+		"first_name": "Jacob",
+		"last_name": "van der Schmidt",
+		"address": "2001 Space Oddysey RD SW Los Angeles, CA",
+		"donations":[
+			("2012-04-20", 20.00),
+			("2019-02-04", 2000.00),
+			("2019-02-04", 3400.00)
+		]
+	},
+	{
+		"first_name": "Jason",
+		"last_name": "Quigley",
+		"address": "12 Camden RD N London, UK",
+		"donations": [
+			("2012-02-04", 12.00),
+			("2013-02-04", 240.00),
+			("2014-02-05",120.00)
+		]
+	},
+	{
+		"first_name": "Ogden",
+		"last_name": "Nash",
+		"address": "256 Camden RD N London, UK",
+		"donations": [
+			("2019-02-04", 5000000.00),
+			("2019-02-04", 500000.00),
+			("2019-02-04", 50000.00)
+		]
+	}
+]
```

Comment:
in this lesson you should be using a dictionary for your data structure, right now you are using a list (to be exactly a list of dictionary items)

an ideal data structure would use a donor name for a key so that your look ups become efficient and this is where we get to practice dictionary feature and why they are useful.

so this is the recommended data structure:
```
donors + {"Ogden Nash": [100, 1002, 299], "Bob Smith": [399]}
```